### PR TITLE
Add stack counter to supporessCommandPort

### PR DESCRIPTION
**Pull request by coleaeason**
@tylerkaraszewski please review. With this change, callers will only be able to reopen the command port if they are the last caller to ask for it to be opened. This means that if the port is already closed when you get to it, unless the original caller that closed it has also tried to reopen, you will not be able to reopen it. I also added some more information to the status output so that we can check on how many callers have asked to close the port, and whether or not the port is currently opened or closed.

Resolves https://github.com/Expensify/Expensify/issues/78929

# Tests
Added new automated testing.

### CODE REVIEW

File: BedrockServer.cpp [hunk 1]
```diff
@@ -1191,10 +1192,12 @@ void BedrockServer::postPoll(fd_map& fdm, uint64_t& nextActivity) {
     }
     if (!_suppressCommandPort && (state == SQLiteNode::MASTERING || state == SQLiteNode::SLAVING) &&
         _shutdownState.load() == RUNNING) {
-        // Open the port
-        if (!_commandPort) {
+        // Open the port if we don't have one and we've had as many opens as we have closures.
+        if (!_commandPort && !_commandPortClosures) {
```
Comment (tylerkaraszewski) [inline BedrockServer.cpp]: This needs to be atomic, something like:
```
int previousClosures = _commandPortClosures.fetch_add(1);
if (previousClosures == 0) {
    // We we're the first one to open this, we'll do the open here.
}
```

File: BedrockServer.cpp [hunk 2]
```diff
@@ -1528,10 +1531,17 @@ void BedrockServer::suppressCommandPort(const string& reason, bool suppress, boo
             closePorts({_controlPort});
             _portPluginMap.clear();
             _commandPort = nullptr;
+            _commandPortClosures += 1;
         }
     } else {
         // Clearing past suppression, but don't reopen (It's always safe to close, but not always safe to open).
         SHMMM("Clearing command port suppression");
+        if (_commandPortClosures) {
```
Comment (tylerkaraszewski) [inline BedrockServer.cpp]: Same as above, make atomic:

```
int previousClosures = _commandPortClosures.fetch_sub(1);
if (previousClosures == 1) {
    // We're the last one to close this (we just set it to 0), we'll do the close
} else if (previousClosures < 1) {
    SWARN("TOO MUCH CLOSING!");
    _commandPortClosures.fetch_add(1);
}
```

**Comment by coleaeason** [comment]
Discussed this with tyler offline some more, decided on a better more atomic way of doing this, so adding a WIP tag

File: BedrockServer.cpp [hunk 3]
```diff
@@ -1521,17 +1522,28 @@ void BedrockServer::suppressCommandPort(const string& reason, bool suppress, boo
...
     } else {
         // Clearing past suppression, but don't reopen (It's always safe to close, but not always safe to open).
         SHMMM("Clearing command port suppression");
+        closures = _commandPortClosures.fetch_sub(1);
+        if (closures == 1) {
+            SINFO("Removing a command port closure.");
+            _commandPort = openPort(_args["-serverHost"]);
+        } else if (closures <= 0) {
+            SWARN("Trying to decrement command port closures past 0, incrementing it back.");
+            closures = _commandPortClosures.fetch_add(1);
```
Comment (tylerkaraszewski) [inline BedrockServer.cpp]: don't need the assignment to `closures` here.

File: BedrockServer.cpp [hunk 4]
```diff
@@ -1746,6 +1759,10 @@ void BedrockServer::_control(BedrockCommand& command) {
         suppressCommandPort("SuppressCommandPort", true, true);
     } else if (SIEquals(command.request.methodLine, "ClearCommandPort")) {
         suppressCommandPort("ClearCommandPort", false, true);
+        int closures = _commandPortClosures.load();
+        if (closures > 1) {
```
Comment (tylerkaraszewski) [inline BedrockServer.cpp]: Why not `>=`?

File: BedrockServer.cpp [hunk 5]
```diff
@@ -1010,9 +1011,9 @@ void BedrockServer::_resetServer() {
 BedrockServer::BedrockServer(const SData& args)
   : SQLiteServer(""), _args(args), _requestCount(0), _replicationState(SQLiteNode::SEARCHING),
     _upgradeInProgress(false), _suppressCommandPort(false), _suppressCommandPortManualOverride(false),
-    _syncThreadComplete(false), _syncNode(nullptr), _suppressMultiWrite(true), _shutdownState(RUNNING),
-    _multiWriteEnabled(args.test("-enableMultiWrite")), _shouldBackup(false), _detach(args.isSet("-bootstrap")),
-    _controlPort(nullptr), _commandPort(nullptr), _maxConflictRetries(3)
+    _commandPortClosures(1), _syncThreadComplete(false), _syncNode(nullptr), _suppressMultiWrite(true),
```
Comment (tylerkaraszewski) [inline BedrockServer.cpp]: Why not `_commandPortClosures(0)`?

**Comment by coleaeason** [comment]
Updated @tylerkaraszewski

**Pull request merged**